### PR TITLE
Add swagger docs restaurant/openhours and restaurant/getall endpoints

### LINKED ISSUE #101: Swagger doc for restaurant get all and get hours endpoints.
**User Story:**
As a backend dev, I should document the APIs so the frontend has easy access to lock up what APIs are available. A swagger doc framework has been set up so this should be fairly easy.

**Acceptance Criteria:**
- [ ] Swagger docs added for /restaurant/getall and /retaurant/openhours endpoints

### PR DESCRIPTION
**Pull request by MingChen91**
## Link to Issue Number
close #101 

## Description
Add swagger docs for /openhours and /getall endpoints.
Fix one error picked up by linter.

## Checklist
- [x] All feature request A/C have been met OR the bug has been fixed
- [x] I ran all necessary tests and they pass
- [x] I rebased upstream/master onto my working branch before opening this PR
- [x] I have named my PR something sensible
- [x] I have included the issue number above
- [x] I have assigned at least two people to review my changes

### CODE REVIEW

**Comment by MingChen91** [comment]
Not 100% sure on swagger syntaxes. Feedback is appreciated.

File: backend/routes/reservation.js [hunk 1]
```diff
@@ -88,7 +88,7 @@ router.get('/all', async (req, res) => {
 // Update a reservation when a user needs to change a field.
 router.put('/:reservationID', async (req, res) => {
   const { date, time, numberOfGuests, notes, firstName, lastName, phoneNumber, email } = req.body;
-  const reservationID = req.params.reservationID;
+  const { reservationID } = req.params;
```
Comment (ljcnz) [inline backend/routes/reservation.js]: what does this actually change?

Comment (ljcnz) [reply backend/routes/reservation.js]: never mind, linter picks this up

**Comment by ljcnz** [comment]
@MingChen91 apologies but I changed a few endpoints in #117 and i believe theres a merge conflict. Could you rebase and slightly change the swagger doc strings? I'll leave review comments specifying where changes need to be made

File: backend/routes/restaurant.js [hunk 2]
```diff
@@ -42,7 +42,24 @@ router.get('/', (req, res) => {
...
+ *
+ * /restaurant/openhours:
+ *   get:
+ *     description: Fetch a restaurant's open hours
+ *     produces:
+ *       - application/json
+ *     parameters:
+ *       - name: restaurantID
+ *         description: Primary Key of Restaurant database table
+ *         in: query
```
Comment (ljcnz) [inline backend/routes/restaurant.js]: in: path

File: backend/routes/restaurant.js [hunk 3]
```diff
@@ -62,7 +79,19 @@ router.get('/openhours', (req, res) => {
   );
 });
 
-// Gets all restaurants ID and name from database. Used to get all available restaurants by front end.
+/**
+ * @swagger
+ *
+ * /restaurant/getall:
```
Comment (ljcnz) [inline backend/routes/restaurant.js]: now just /restaurant

File: backend/routes/restaurant.js [hunk 4]
```diff
@@ -42,7 +42,24 @@ router.get('/', (req, res) => {
   });
 });
 
-// Gets specific restaurant opening hours, used by front end to build restaurant opening hours for end user.
+/**
+ * @swagger
+ *
+ * /restaurant/openhours:
```
Comment (ljcnz) [inline backend/routes/restaurant.js]: now restaurant/{id}/openhours

File: backend/routes/restaurant.js (hunk 2 above)
Comment (MingChen91) [reply backend/routes/restaurant.js]: done

File: backend/routes/restaurant.js (hunk 3 above)
Comment (MingChen91) [reply backend/routes/restaurant.js]: done

File: backend/routes/restaurant.js (hunk 4 above)
Comment (MingChen91) [reply backend/routes/restaurant.js]: done

**Comment by MingChen91** [comment]
rebased and fixed swagger docs . @ljcnz . thanks for making the apis RESTful. Made changes accordingly now

**Comment by itsercleung** [comment]
We've just pulled your swagger doc changes to Nidhi's PR and have merged - We will credit you Ming

**Pull request closed without merging**